pylint/utils/utils.py

```diff
@@ -148,7 +170,58 @@ def register_plugins(linter, directory):
                     imported[base] = 1
 
 
-def get_global_option(checker, option, default=None):
+@overload
+def get_global_option(
+    checker: "BaseChecker", option: GLOBAL_OPTION_BOOL, default: Optional[bool] = None
+) -> bool:
+    ...
+
+
+@overload
+def get_global_option(
+    checker: "BaseChecker", option: GLOBAL_OPTION_INT, default: Optional[int] = None
+) -> int:
+    ...
+
+
+@overload
+def get_global_option(
+    checker: "BaseChecker",
+    option: GLOBAL_OPTION_LIST,
+    default: Optional[List[str]] = None,
+) -> List[str]:
+    ...
+
+
+@overload
+def get_global_option(
+    checker: "BaseChecker",
+    option: GLOBAL_OPTION_PATTERN,
+    default: Optional[Pattern] = None,
+) -> Pattern:
+    ...
+
+
+@overload
+def get_global_option(
+    checker: "BaseChecker",
+    option: GLOBAL_OPTION_TUPLE_INT,
+    default: Tuple[int, ...] = None,
+) -> Tuple[int, ...]:
+    ...
+
+
+def get_global_option(
+    checker: "BaseChecker",
+    option: Union[
+        GLOBAL_OPTION_BOOL,
+        GLOBAL_OPTION_INT,
+        GLOBAL_OPTION_LIST,
+        GLOBAL_OPTION_PATTERN,
+        GLOBAL_OPTION_TUPLE_INT,
+    ],
+    default: Union[None, bool, int, List[str], Pattern, Tuple[int, ...]] = None,
+) -> Union[None, bool, int, List[str], Pattern, Tuple[int, ...]]:
```

Comment:
We could define a type alias for it as well.

--
Let's also try something different. I haven't checked if it would work out,  but what about using a `TypeVar`?
```py
from typing import TypeVar

T = TypeVar("T", bool, int, List[str], Pattern, Tuple[int, ...])

def get_global_options(checker: ..., option: ..., default: Optional[T] = None) -> T:
    ...
```